Assets/Scripts/Game/Items/LootTables.cs

```diff
@@ -166,15 +166,15 @@ public static DaggerfallUnityItem[] GenerateRandomLoot(LootChanceMatrix matrix,
 
             // Random weapon
             chance = matrix.WP;
-            while (Random.Range(0, 100) < chance)
+            while (Random.Range(0, 100 + 1) < chance)
             {
                 items.Add(ItemBuilder.CreateRandomWeapon(playerEntity.Level));
                 chance *= 0.5f;
             }
 
             // Random armor
             chance = matrix.AM;
-            while (Random.Range(0, 100) < chance)
+            while (Random.Range(0, 100 + 1) < chance)
             {
```

Comment:
I think these should be as they are, in order for the percent chances of the items spawning to work as expected. For example, in order for 100% spawn chance to work 100% of the time.

Reply:
You're right, if "chance" is a percentage of success, then a correct way to check success is
Random.Range(0, 100) < chance
or Random.Range(1, 101) <= chance

and consequently, to check for failure,
Random.Range(0, 100) >= chance
or Random.Range(1, 101) > chance

Only checks that do not match those cases are suspicious.
Ok, I need more time to work on that.